authentication/typescript/azure/src/azureIdentityAccessTokenProvider.ts

```diff
@@ -1,22 +1,35 @@
 import { GetTokenOptions, TokenCredential } from "@azure/core-auth";
-import { AccessTokenProvider } from '@microsoft/kiota-abstractions';
+import { AccessTokenProvider, AllowedHostsValidator } from '@microsoft/kiota-abstractions';
 
 export class AzureIdentityAccessTokenProvider implements AccessTokenProvider {
     /**
      *@constructor
      *@param credentials The tokenCredential implementation to use for authentication.
      *@param scopes The scopes to use for authentication.
      *@param options The options to use for authentication.
+     *@param allowedHosts The allowed hosts to use for authentication.
      */
-    public constructor(private readonly credentials: TokenCredential, private readonly scopes: string[] = ['https://graph.microsoft.com/.default'], private readonly options?: GetTokenOptions) {
+    public constructor(private readonly credentials: TokenCredential, 
+        private readonly scopes: string[] = ['https://graph.microsoft.com/.default'],
+        allowedHosts: string[] = ['graph.microsoft.com', 'graph.microsoft.us', 'dod-graph.microsoft.us', 'graph.microsoft.de', 'microsoftgraph.chinacloudapi.cn', 'canary.graph.microsoft.com'],
```

Comment:
This list of hosts will be moving to Graph core right?

Reply:
Yeah. @baywet created https://github.com/microsoft/kiota/issues/1063 for this